inventory/inventory_test.go

```diff
@@ -137,3 +146,705 @@ func TestWrite(t *testing.T) {
 		t.Errorf("writeInventory call did not write %q", k)
 	}
 }
+
+func TestInventoryInfoRpm(t *testing.T) {
+	var AppFs= afero.NewMemMapFs()
+	assertion := assert.New(t)
+	releaseFile := "/etc/os-release"
+	AppFs.Create(releaseFile)
+	fcontent := `PRETTY_NAME="Debian buster"
+NAME="Debian GNU/Linux"
+VERSION_ID="10"
+VERSION="10 (buster)"
+VERSION_CODENAME=buster
+ID=debian
+HOME_URL="https://www.debian.org/"
+SUPPORT_URL="https://www.debian.org/support"
+BUG_REPORT_URL="https://bugs.debian.org/"
+`
+
+	rpmQueryResult := `foo x86_64 1.2.3-4
+`
+	afero.WriteFile(AppFs, releaseFile, []byte(fcontent), 644)
+	AppFs.Create("/usr/bin/rpmquery")
+
+	stubs := gostub.Stub(&common.OsHostname, func() (string, error) {
+		return "test-hostname", nil
+	})
+
+	stubs.Stub(&osinfo.GetUname, func() ([]byte, error) {
+		return []byte("test-kernel"), nil
+	})
+
+	stubs.Stub(&common.ReadFile, func(file string) ([]byte, error) {
+		return []byte(fcontent), nil
+	})
+
+	stubs.Stub(&common.Exists, func(name string) bool {
+		if _, err := AppFs.Stat(name); err != nil {
+			return false
+		}
+		return true
+	})
+
+	stubs.Stub(&packages.ZypperExists, false)
+	stubs.Stub(&packages.AptExists, false)
+	stubs.Stub(&packages.GemExists, false)
+	stubs.Stub(&packages.PipExists, false)
+	stubs.Stub(&packages.YumExists, false)
+
+	stubs.Stub(&osinfo.Architecture, func(arch string) string {
+		return "x86_64"
+	})
+
+	stubs.Stub(&common.Run, func(cmd *exec.Cmd, logger *log.Logger) ([]byte, error) {
+		return []byte(rpmQueryResult), nil
+	})
+	defer stubs.Reset()
+
+	config.SetVersion("1")
+
+	ii := Get()
+
+	assertion.Equal("test-hostname", ii.Hostname, "hostname does not match")
+	assertion.Equal("Debian buster", ii.LongName, "long name does not match")
+	assertion.Equal("test-kernel", ii.KernelVersion, "kernel version does not match")
+	assertion.Equal("foo", ii.InstalledPackages.Rpm[0].Name, "package name does not match")
+	assertion.Equal("1.2.3-4", ii.InstalledPackages.Rpm[0].Version, "package version does not match")
```

Comment:
Should there be some tests that do match?

Reply:
the tests are for matching. the messages are error messages if the assertion fails.